src/main/java/com/ecore/roles/model/Role.java

```diff
@@ -29,6 +30,7 @@ public class Role {
     private UUID id;
 
     @Column(nullable = false, unique = true)
+    @NotBlank(message = "Name must not be empty")
```

Comment:
I think it makes more sense to use Bean Validation instead of throwing InvalidArgumentException. The code becomes cleaner with its annotations: @NotNull, @NotEmpty, etc...